ДЗ 9: Самодельный ORM

anv0l opened the pull request:
Цель:
Научиться работать с jdbc. На практике освоить многоуровневую архитектуру приложения.

Работа должна использовать базу данных в docker-контейнере .

В модуле homework реализуйте классы:

EntityClassMetaData
EntitySQLMetaData
DataTemplateJdbc
Метод main в классе HomeWork должен работать без ошибок.

anv0l committed:
ДЗ 9: Самодельный ORM

Цель:
Научиться работать с jdbc. На практике освоить многоуровневую архитектуру приложения.

Работа должна использовать базу данных в docker-контейнере .

В модуле homework реализуйте классы:

EntityClassMetaData
EntitySQLMetaData
DataTemplateJdbc
Метод main в классе HomeWork должен работать без ошибок.

petrelevich commented on hw09-jdbc/homework/build.gradle:
```diff
@@ -0,0 +1,19 @@
+
+dependencies {
+    dependencies {
+        implementation project(':hw09-jdbc:demo')
```
для модуля hw09-jdbc:demo не хватате build.gradle.

petrelevich commented on hw09-jdbc/homework/src/main/java/ru/otus/jdbc/mapper/EntityClassMetaDataImpl.java:
```diff
@@ -0,0 +1,88 @@
+    private Field getFieldId(Field[] declaredFields) {
+        Field found = null;
+        for (Field declaredField : declaredFields) {
+            if (declaredField.isAnnotationPresent(PRIMARY_KEY_ID)) {
```
тут хорошо stream api применить.

petrelevich commented on hw09-jdbc/homework/src/main/java/ru/otus/jdbc/mapper/EntityClassMetaDataImpl.java:
```diff
@@ -0,0 +1,88 @@
+        try {
+            constructor = clazz.getConstructor();
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
```
если конструктор не найден, то стоит выбрасывать ошибку.

anv0l committed:
Исправления:
1. Переделал на stream API.
2. Вернул забытый build.gradle.
3. Если не найден конструктор - выбрасываем исключение.

anv0l committed:
Merge branch 'main' into hw09-jdbc
